conda_build/inspect_pkg.py

```diff
@@ -67,43 +67,21 @@ def which_package(
     Given the path (of a (presumably) conda installed file) iterate over
     the conda packages the file came from.  Usually the iteration yields
     only one package.
-
-    We use lstat since a symlink doesn't clobber the file it points to.
     """
-    prefix = Path(prefix)
-
-    # historically, path was relative to prefix, just to be safe we append to prefix
-    # get lstat before calling _file_package_mapping in case path doesn't exist
     try:
-        lstat = (prefix / path).lstat()
-    except FileNotFoundError:
-        # FileNotFoundError: path doesn't exist
-        return
-    else:
-        yield from _file_package_mapping(prefix).get(lstat, ())
-
+        path = Path(path).relative_to(prefix)
+    except ValueError:
+        # If path is already relative to prefix, we get a ValueError.
+        pass
+    # On Windows, be lenient and allow case-insensitive path comparisons.
+    # NOTE: On macOS, although case-insensitive filesystem is default, still
+    #       require case-sensitive matches (i.e., normcase on macOS is a no-op).
+    normcase_path = normcase(str(path).replace(os.sep, "/"))
```

Comment:
In the current 3.28 versions, there was a behavioral change in that builds on a case-insensitive filesystem on Unix systems (i.e., macOS with default file system) would ignore case mismatches.
Pre-3.28 behavior was to check for case-sensitivity on all Unixes (since `normcase`'s implementation only ignores cases on Windows).

Reply:
from Python docs on `os.path.normcase`:
> On Windows, convert all characters in the pathname to lowercase, and also convert forward slashes to backward slashes.

```suggestion
    # On Windows, be lenient and allow case-insensitive path comparisons.
    # NOTE: On macOS, although case-insensitive filesystem is default, still
    #       require case-sensitive matches (i.e., normcase on macOS is a no-op).
    normcase_path = normcase(path)
```